app/models/direct_award.py

```diff
@@ -1,19 +1,23 @@
 from datetime import datetime
 from urllib.parse import urljoin
 
+from sqlalchemy.dialects.postgresql import BIGINT
 from sqlalchemy.orm import validates
 from flask import current_app
 
 from app import db
+from app.utils import random_positive_external_id
 from app.url_utils import force_relative_url
 from app.models import User, ValidationError, ArchivedService
+
 from dmutils.formats import DATETIME_FORMAT
 
 
 class DirectAwardProject(db.Model):
     __tablename__ = 'direct_award_projects'
 
     id = db.Column(db.Integer, primary_key=True)
+    external_id = db.Column(BIGINT, default=random_positive_external_id, nullable=False, index=True, unique=True)
```

Comment:
For supplier ID we use `db.BigInteger` - I'm pretty sure these will be the same thing under the hood, but is it worth using that here for consistency?
```
    supplier_id = db.Column(db.BigInteger,
                            db.ForeignKey('suppliers.supplier_id'),
                            index=True, unique=False, nullable=True)
```